### DISCUSSION TITLE
Redirects, Server Middleware and the Nuxt context

### DISCUSSION THREAD

**Discussion by elHornair** [nuxt/nuxt#16906] [Questions] [answered]
I need to do server side HTTP redirects (not just vue internal redirects). In Nuxt 2, there seemed to be a [redirect method](https://nuxtjs.org/docs/internals-glossary/context/#redirect) specifically for this. According to [this](https://github.com/nuxt/framework/discussions/906), redirects can be done in a Server Middleware. For generating the target URL, I need the router and thus the Nuxt context. How can I access the Nuxt context in a server middleware?

**Comment by Diizzayy** [comment, marked as answer]
HTTP redirects can be achieved by using the res ( ServerResponse ) in ssrContext as seen below

```js
  const { ssrContext } = useNuxtApp();
  const { res } = ssrContext;

  res.writeHead(302, { Location: "/redirect-to" });
  res.end();
```

**Comment by torressam333** [comment]
@Diizzayy  you da man! This is the last piece of a complex puzzle I was working on.

**Reply by Diizzayy** [reply]
@torressam333 Happy to hear that this helped you out!
